src/commons/sideContent/SideContentContestLeaderboard.tsx

```diff
@@ -26,15 +28,56 @@ const columnHeader = (colClass: string, colTitle: string) => (
   </div>
 );
 
-const contestEntryHeader = (
-  <div className="leaderboard-header">
-    {columnHeader('header-entryid', 'Student Name')}
-    {columnHeader('header-entryrank', 'Rank')}
-    {columnHeader('header-score', 'Score')}
-  </div>
-);
+const contestEntryHeader = (leaderboardType: SideContentType) => {
+  let leaderboardMetric = '';
+  switch (leaderboardType) {
+    case SideContentType.scoreLeaderboard:
+      leaderboardMetric = 'Calculated Score';
+      break;
+    case SideContentType.popularVoteLeaderboard:
+      leaderboardMetric = 'Popularity Score';
+      break;
+    default:
+      leaderboardMetric = 'Metric';
+  }
+  return (
+    <div className="leaderboard-header">
+      {columnHeader('header-entryid', 'Student Name')}
+      {columnHeader('header-entryrank', 'Rank')}
+      {columnHeader('header-score', leaderboardMetric)}
+    </div>
+  );
+};
+
+const leaderboardTitle = (leaderboardType: SideContentType) => {
+  let leaderboardTitle = '';
+  switch (leaderboardType) {
+    case SideContentType.scoreLeaderboard:
+      leaderboardTitle = 'Score Leaderboard';
+      break;
+    case SideContentType.popularVoteLeaderboard:
+      leaderboardTitle = 'Popular Vote Leaderboard';
+      break;
+    default:
+      leaderboardTitle = 'Contest Leaderboard';
+  }
+  return leaderboardTitle;
+};
```

Comment:
Minor nit: I prefer to use a ternary ladder here, it avoids the use of `let` and it also avoids creating an additional function, and we can define it inline (or even memoize using `useMemo`):

```ts
const leaderboardTitle = leaderboardType === SideContentType.scoreLeaderboard
    ? 'Score Leaderboard'
    : leaderboardType === SideContentType. popularVoteLeaderboard
    ? 'Popular Vote Leaderboard'
    : 'Contest Leaderboard';
```

What do you think? Similarly for the other values.

Reply:
Yep, that makes sense. I've refactored the code so that avoids using switch cases and I've also added useMemo to make it a bit better.